distributed/spans.py

```diff
@@ -262,8 +265,25 @@ def nbytes_total(self) -> int:
         """
         return sum(tg.nbytes_total for tg in self.traverse_groups())
 
+    @property
+    def cumulative_worker_metrics(self) -> defaultdict[tuple[str, ...], float]:
+        """Replica of Worker.digests_total and Scheduler.cumulative_worker_metrics, but
+        only for the metrics that can be attributed to the current span tree.
+        The span_id has been removed from the key.
+
+        At the moment of writing, all keys are
+        ``("execute", <task prefix>, <activity>, <unit>)``
+        but more may be added in the future with a different format; please test for
+        ``k[0] == "execute"``.
```

Comment:
Do you think it'd be possible later on to map everything to some dataclass or similar? There's an awful lot of tedious checks building up for key lengths / values to determine what's what.

Reply:
it is certainly possible and sensible, yes